### PR TITLE
Added Annotations for PyTorch models

### PR DESCRIPTION
**Pull request by anmolsjoshi**
# What does this PR do?

Adds annotations as described in https://github.com/huggingface/transformers/issues/16059:
* Updated type annotations for CTRL (PT), RAG(PT)
* added type annotations for FSMT(PT), LED(PT), M2M(PT), MPNet(PT) , Nystromformer(PT), OpenAI(PT)

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@sgugger

### CODE REVIEW

**Comment by anmolsjoshi** [comment]
@sgugger not sure why the tests are failing. I'm pretty sure they aren't related to added annotations

File: src/transformers/models/led/modeling_led.py [hunk 1]
```diff
@@ -1572,6 +1572,10 @@ class LEDSeq2SeqQuestionAnsweringModelOutput(ModelOutput):
             you can choose to directly pass an embedded representation. This is useful if you want more control over
             how to convert `input_ids` indices into associated vectors than the model's internal embedding lookup
             matrix.
+        inputs_embeds (`torch.FloatTensor` of shape `(batch_size, sequence_length, hidden_size)`, *optional*):
```
Comment (sgugger) [inline src/transformers/models/led/modeling_led.py:1575]: They are actually in the paragraph above, just badly formatted. Could you remove them from there?

**Review by sgugger — APPROVED** [review]
Thanks for adding all of those!

File: src/transformers/models/led/modeling_led.py (hunk 1 above)
Comment (anmolsjoshi) [reply src/transformers/models/led/modeling_led.py]: @sgugger thanks for the review. I deleted that docstring and corrected the "badly formatted" one.

**Comment by sgugger** [comment]
Thanks! We just need a last `make style` on your branch for the code quality check :-)

**Pull request merged**